### PR TITLE
Add branch and create functionality

### PR DESCRIPTION
**Pull request by amanuelbelete95**
## List OF Completed Activity
- Wrapped  App component with the Provider component from react-redux
- Passed your configured store into the <Provider>
- Display your books, received from the slice, in a reusable component
- Dispatch actions using useDispatch
- Added a Button component, which includes:
- Added an event handler that adds a book to the books array (with attributes id, title and author)
- Added a Button component, which includes:
- Added An event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/components/BookList.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                gap: '10px',
+                marginTop: '25px',
+              }}
+              >
+                <button type="submit">Comments</button>
+                <button onClick={() => dispatch(deleteBook(book.item_id))} type="submit">Remove</button>
+                <button type="submit">Edit</button>
+              </div>
+            </div>
+            <div style={{
```
Comment (victorokoroji) [inline src/components/BookList.jsx:37]: - [ ] To make your codebase clean and your components reusable, i will recommend that you create a separate component called <Book /> which will contain a single book item. You can then import it into your <BookList /> component where you will then map to display all your books. This way, your code will be modular and readable.

File: src/components/Books.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            required
+          />
+          <select
+            value={author}
+            onChange={(e) => setCat(e.target.value)}
+          >
+            <option value="action">Fiction</option>
+            <option value="sci-fi">Nonfiction</option>
+            <option value="Economy">Drama</option>
+          </select>
```
Comment (victorokoroji) [inline src/components/Books.jsx:49]: - [ ] To make your codebase clean and your components reusable, I will recommend that you create a separate component called <NewBookForm /> which will contain the form to add a book.  This way, your code will be modular and readable.

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @amanuelbelete95 , 

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- Redux is properly setup ✅ 
- The PR title and description are professional and descriptive ✅ 
- Redux actions are working as expected ✅ 



## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @amanuelbelete95, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- `initialState` has been added to the `booksSlice` ✔️
- Has `<Provider>` component from `react-redux` ✔️
- Has `<BookList>` with `useSelector` ✔️
- Has a Component for Adding and Removing Books ✔️
- Good use of GitFlow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**